ranger-companion.lic

```diff
@@ -0,0 +1,68 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#ranger-companion
+=end
+
+custom_require.call(%w[common])
+
+no_pause_all
+
+class Companion
+  include DRC
+
+  def initialize
+    # TODO: Add raccoon messaging
+    case bput('whistle for companion', 'Your lips are too dry', 'wolf scrambles in', 'You whistle a merry tune', 'wolf perks', 'You whistle loudly for your companion', 'purse your lips to call')
+    when 'purse your lips to call'
+      exit
+    end
+
+    while line = get
+      waitrt?
+      if line =~ /^.*(young|full-grown) wolf (begins|howls|paws|looks|perks|sits|pants|stands|carefully)/
+        bput('signal wolf to hunt', 'suddenly catches a scent', 'is too young to hunt', 'Your companion is not', 'You have no companion', 'snapping your fingers')
+        exit
+      end
+      if line =~ /^.*paces back and forth*/
+        case bput('pet wolf', 'You pet your baby wolf', 'Touch what', 'wolf shies away from you')
+        when 'wolf shies away from you'
+          bput('pet second wolf', 'You pet your baby wolf', 'Touch what', 'wolf shies away from you')
+        end
+        bput('pet raccoon', 'You pet', 'Touch what')
```

Comment:
Is the intent to always `pet wolf` and `pet raccoon` consecutively?

Reply:
yeah I just added that because the regex matches both types pacing, and I don't know the exact messages for raccoons. I will narrow this down now though.